services/user.py

```diff
@@ -0,0 +1,43 @@
+from django.shortcuts import get_object_or_404
+
+from db.models import User
+
+
+def create_user(username: str, password: str, email: str = None,
+                first_name: str = None, last_name: str = None) -> User:
+    if User.objects.filter(username=username).exists():
+        raise ValueError("A user with this username already exists.")
+    if email and User.objects.filter(email=email).exists():
+        raise ValueError("A user with this email already exists.")
+    user = User.objects.create_user(username=username, password=password)
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+    user.save()
+    return user
+
+
+def get_user(user_id: int) -> User:
+    return get_object_or_404(User, id=user_id)
+
+
+def update_user(user_id: int, username: str = None, password: str = None, email: str = None,
+                first_name: str = None, last_name: str = None) -> User:
+    user = User.objects.get(id=user_id)
+
+    if username:
+        user.username = username
+    if password:
+        user.set_password(password)
+    if email:
+        user.email = email
```

Comment:
Before updating the user, ensure that the new username and email are unique. This is crucial to prevent data integrity issues.